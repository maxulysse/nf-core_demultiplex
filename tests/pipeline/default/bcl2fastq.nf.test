nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Bcl2Fastq") {

        when {
            params {
                outdir = "$outputDir"
                input = "${baseDir}/assets/inputs/flowcell_input.csv"
                demultiplexer = 'bcl2fastq'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_data.txt"),
                                  path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001.fastp.json"),
                                  path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_summary.txt"),
                                  path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001.fastp.fastq.gz"),
                                  path("$outputDir/bcl2fastq/Sample1_S1_L001_R1_001.fastq.gz"),
                                  path("$outputDir/bcl2fastq/Stats/AdapterTrimming.txt"),
                                  path("$outputDir/bcl2fastq/Stats/ConversionStats.xml"),
                                  path("$outputDir/bcl2fastq/Stats/DemultiplexingStats.xml"),
                                  path("$outputDir/bcl2fastq/Stats/DemuxSummaryF1L1.txt"),
                                  path("$outputDir/bcl2fastq/Stats/FastqSummaryF1L1.txt"),
                                  path("$outputDir/bcl2fastq/Stats/Stats.json"),
                                  path("$outputDir/bcl2fastq/220422_M11111_0222_000000000-K9H97/InterOp/ControlMetricsOut.bin"),
                                  path("$outputDir/bcl2fastq/220422_M11111_0222_000000000-K9H97/InterOp/CorrectedIntMetricsOut.bin"),
                                  path("$outputDir/bcl2fastq/220422_M11111_0222_000000000-K9H97/InterOp/ErrorMetricsOut.bin"),
                                  path("$outputDir/bcl2fastq/220422_M11111_0222_000000000-K9H97/InterOp/ExtractionMetricsOut.bin"),
                                  path("$outputDir/bcl2fastq/220422_M11111_0222_000000000-K9H97/InterOp/IndexMetricsOut.bin"),
                                  path("$outputDir/bcl2fastq/220422_M11111_0222_000000000-K9H97/InterOp/QMetricsOut.bin"),
                                  path("$outputDir/bcl2fastq/220422_M11111_0222_000000000-K9H97/InterOp/TileMetricsOut.bin"),).match("bcl2fastq") },
                { assert new File("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001.fastp.html").exists() },
                { assert new File("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_report.html").exists() }                     
            )
        }

    }

}
