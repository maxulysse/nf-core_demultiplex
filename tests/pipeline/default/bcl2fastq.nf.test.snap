{
    "software_versions": {
        "content": [
            "software_versions.yml:md5,b3d4b4876965c4097a3fb31e441bab6b"
        ],
        "timestamp": "2023-04-06T06:22:17+0000"
    },
    "bcl2fastq": {
        "content": [
            "Sample1_S1_L001_data.txt:md5,0adf4e2617f55fabba0663583c1f65b7",
            "Sample1_S1_L001.fastp.json:md5,91fcd16615456aba380203a6ee2dad59",
            "Sample1_S1_L001_summary.txt:md5,510c6b8a26ff8d530950c05b3ca5a1c6",
            "Sample1_S1_L001.fastp.fastq.gz:md5,081f3075bdc3f3f410058035124f1eb2",
            "Sample1_S1_L001_R1_001.fastq.gz:md5,e92fce7b86c6447b053d72c5cb4be89c",
            "AdapterTrimming.txt:md5,48ed2b914b1246c0b5d8667525550946",
            "ConversionStats.xml:md5,8fe0f57f3f5d256a0762dba75ac62d05",
            "DemultiplexingStats.xml:md5,2047ff18f5b9107c084de06e9ff943ad",
            "DemuxSummaryF1L1.txt:md5,03e5fd0c1e3079c5f8c7b4d0501b37ff",
            "FastqSummaryF1L1.txt:md5,0c6f2d87ee183b84d1051cde9a5643d1",
            "Stats.json:md5,8e5f038b8aa9e465599d3575f930e604",
            "ControlMetricsOut.bin:md5,6d77b38d0793a6e1ce1e85706e488953",
            "CorrectedIntMetricsOut.bin:md5,2bbf84d3be72734addaa2fe794711434",
            "ErrorMetricsOut.bin:md5,38c88def138e9bb832539911affdb286",
            "ExtractionMetricsOut.bin:md5,7497c3178837eea8f09350b5cd252e99",
            "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
            "QMetricsOut.bin:md5,7e9f198d53ebdfbb699a5f94cf1ed51c",
            "TileMetricsOut.bin:md5,83891751ec1c91a425a524b476b6ca3c"
        ],
        "timestamp": "2023-04-06T06:22:17+0000"
    }
}